main.cpp

```diff
@@ -121,357 +163,390 @@ struct CarWash
+}
+
+void Filter::trackPitch(float cv)
+{
+    cutoff += cv;
+}
+
+void Filter::modFilter(float cv)
+{
+    //fmAttenuator works between values of 0.0f and 1.0f
+    cutoff += cv * fmAttenuator;
+    
+}
+
+struct Envelope
 {
-    //1)  attack time in ms
     int attack = 243;
-    //2)  decay time in ms
     int decay = 501;
-    //3)  sustain level in percentage
     float sustain = 0.67f;
-    //4)  release time in ms
     int release = 684;
-    //5)  hold time in ms
     int hold = 43;
 
-    //1)  output CV
-    float outputCV(bool powerOn, bool gateHigh);
-    //2)  receive gate/trigger
-    bool receive(bool cycleModeOnOff = false);
-    //3)  engage cycle mode
-    bool cycleMode(bool switchPosition);
+    bool willOutputCV(bool powerOn, bool gateHigh);
+    bool receiveGate(bool cycleModeOn = false);
+    void cycleMode(bool switchOn);
 };
 
-/*
-Thing 8) VCAs
-5 properties:
-    1)  number of audio inputs
-    2)  number of cv inputs
-    3)  cascading cv inputs (bool)
-    4)  boost switch
-    5)  number of audio outputs
-3 things it can do:
-    1)  receive cv
-    2)  output audio/cv
-    3)  mute output
- */
-struct VCAs FIXME: is this representing ONE VCA or multiple VCAs?
-{
-    //1)  number of audio inputs
-    int audioInputs = 4;
-    //2)  number of cv inputs
-    int cvInputs = 4;
-    //3)  cascading cv inputs
-    bool cascadingCV = true;
-    //4)  boost switch
-    bool boostSwitch = true;
-    //5)  number of audio outputs
-    int audioOutputs = 4;
+bool Envelope::willOutputCV(bool powerOn, bool gateHigh)
+{
+    if (powerOn == true && gateHigh == true)
+    {
+        return true;
+    }
+    else
+    {
+        return false;
+    }
+}
+
+bool Envelope::receiveGate(bool cycleModeOn)
+{
+    if (cycleModeOn == false)
+    {
+        return true;
+    }
+    else
+    {
+        return false;
+    }
+}
 
+void Envelope::cycleMode(bool switchOn)
+{
+    if (switchOn == true)
+    {
+        switchOn = false;
+    }
+    else
+    {
+        switchOn = true;
+    }
+}
 
-    //1)  receive cv
-    void receiveCV(int cvInputs);
-    //2)  output audio/cv
-    void output();
-    //3)  mute output
-    void mute(bool muteSwitch);
+struct DualVCA
+{
+    int audioInputs = 2;
+    int cvInputs = 2;
+    bool cvInputUsed = false;
+    bool boostSwitch = true;
+    int audioOutputs = 2;
+    float cvInputOne = 12.02f;
+    float cvInputTwo = 0.0f;
+
+    void cascadeCV();
+    void boost();
+    void mute(bool muteOn);
 };
+void DualVCA::cascadeCV()
+{
+    for (int i = 0; i <= cvInputs; ++i)
+    {
+        if (cvInputUsed == false)
+        {
+            cvInputTwo = cvInputOne;
+        }
+    }
+}
+
+
+void DualVCA::boost()
+{
+    if (boostSwitch == false)
+    {
+        boostSwitch = true;
+    }
+    else
+    {
+        boostSwitch = false;
+    }
+}
+
+void DualVCA::mute(bool muteOn)
+{
+    if (muteOn == true)
+    {
+        muteOn = false;
+    }
+    else
+    {
+        muteOn = true;
+    }
+}
```

Comment:
```
muteOn = !muteOn;
```
but you're just toggling the state of the function parameter.  so maybe add a member variable that you'll toggle the state of?